# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1237814628 -3600
#      Mon Mar 23 14:23:48 2009 +0100
# Node ID 7e31df9a03347af359fedf8fa25cc9e79446edc1
# Parent  979d0492ff4054e3fa55289b952a4173b04ed388
simplify double->int conversion in indexing

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2009-03-23  Jaroslav Hajek  <highegg@gmail.com>
+
+	* idx-vector.cc (convert_index(double,...)): Simplify.
+
 2009-03-21  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Array-d.cc: lo_ieee_isnan -> xisnan.
 	* Array-f.cc: Ditto.
 	* oct-inttypes.cc: Ditto.
 	* oct-inttypes.h: Ditto.
 	* CDiagMatrix.cc: Add missing include.
 	* fCDiagMatrix.cc: Ditto.
diff --git a/liboctave/idx-vector.cc b/liboctave/idx-vector.cc
--- a/liboctave/idx-vector.cc
+++ b/liboctave/idx-vector.cc
@@ -177,28 +177,19 @@ convert_index (octave_idx_type i, bool& 
     conv_error = true;
   if (ext < i) ext = i;
   return i - 1;
 }
 
 inline octave_idx_type
 convert_index (double x, bool& conv_error, octave_idx_type& ext)
 {
-  octave_idx_type i;
-  if (xisnan (x) || xisinf (x))
-    {
-      i = 0;
-      conv_error = true;
-    }
-  else
-    {
-      i = static_cast<octave_idx_type> (x);
-      if (static_cast<double> (i) != x)
-        conv_error = true;
-    }
+  octave_idx_type i = static_cast<octave_idx_type> (x);
+  if (static_cast<double> (i) != x)
+    conv_error = true;
 
   return convert_index (i, conv_error, ext);
 }
 
 inline octave_idx_type
 convert_index (float x, bool& conv_error, octave_idx_type& ext)
 {
   return convert_index (static_cast<double> (x), conv_error, ext);
